src/main/java/com/github/leroyguillaume/keycloak/bcrypt/BCryptPasswordHashProviderFactory.java

```diff
@@ -29,7 +29,7 @@
  */
 public class BCryptPasswordHashProviderFactory implements PasswordHashProviderFactory {
     public static final String ID = "bcrypt";
-    public static final int DEFAULT_ITERATIONS = 8192;
+    public static final int DEFAULT_ITERATIONS = 10;
```

Comment:
@leroyguillaume With the code that I had added in the previous PR, this would result in an insecure number of iterations for a provider that uses the default. 

I had adapted the previous PR to use the actual number of iterations that the salt generation function should target as an input. The approach was to convert that to log2(number of iterations) which the underlying library expects. 

With this default you will endup with 4 log rounds iterations because `iterationsToLogRounds` is used. The current recommended standard is at least 12 

One approach would be to document that the provider is configured to use log2 of iterations. In that case, the code will need to be modified to remove the use of `iterationsToLogRounds`. 
I had added the adapter to preserve compatibility with existing deployments that may already be in the wild and using the full number of iterations. 

If you prefer to switch the model, I would be happy to submit a PR, although, my concern would be affecting existing deployments.

Reply:
Oh... I see the problem. 

You can submit a PR to switch the model if you want to, it's very nice of you to consider this project!
For existing deployment, we can precise breaking change in release changelog and increase major version. What do you think?

Reply:
Thanks for being so responsive. We are using the bcrypt provider since we are migrating an existing user base to keycloak that has bcrypt password hashes, so the least we can do is contribute. 

For this particular issue, I would actually probably recommend sticking with a user facing value that is the full number of iterations. Seems consistent with how Keycloak treats other hash providers. It can be a bit confusing, but we can just describe in the documentation how to properly set the complexity factor. 

FYI, we are moving to Keycloak 8.0.1 which had some breaking changes to to the interfaces so we would submit a PR shortly to address that.